src/embedded_flash/upgrade_storage.rs

```diff
@@ -48,4 +48,9 @@ pub trait UpgradeStorage {
     ///
     /// Returns [`StorageError::OutOfBounds`] if the data is too long to fit the metadata storage.
     fn write_metadata(&mut self, data: &[u8]) -> StorageResult<()>;
+
+    /// Returns an identifier for these upgrade locations.
```

Comment:
This is not clear to me. What are "these" upgrade locations? As I understand it we have the following distinct locations: Partition A, Partition B, Metadata. Does this mean "the current running partition" or "the next writable partition".

Reply:
There is one metadata location per partition. And you only have access to the partition you are not currently on yourself. "these" are the ones in `self`. Maybe the later comments will also help clear things up.

Reply:
Yeah, I was missing the fact that each "side" (A/B) has its own metadata.